src/main/scala/mesosphere/marathon/api/v2/validation/PodsValidation.scala

```diff
@@ -0,0 +1,162 @@
+package mesosphere.marathon.api.v2.validation
+
+// scalastyle:off
+import java.util.regex.Pattern
+
+import com.wix.accord.dsl._
+import com.wix.accord.{ Failure, Result, RuleViolation, Success, Validator }
+import mesosphere.marathon.Features
+import mesosphere.marathon.api.v2.Validation
+import mesosphere.marathon.raml.{ Constraint, EnvVars, FixedPodScalingPolicy, MesosContainer, Network, Pod, PodPlacementPolicy, PodScalingPolicy, PodSchedulingBackoffStrategy, PodSchedulingPolicy, PodUpgradeStrategy, Resources, Secrets, Volume }
+import mesosphere.marathon.state.{ PathId, ResourceRole }
+
+import scala.collection.immutable.Seq
+import scala.util.Try
+// scalastyle:on
+
+/**
+  * Defines implicit validation for pods
+  */
+trait PodsValidation {
```

Comment:
No, it's a trait so it can be mixed in and any implicits are lower priority. The companion object extends it...